frontend/src/diagram/sprotty/DependencyInjection.ts

```diff
@@ -66,27 +68,66 @@ import {
 } from 'sprotty';
 import { Action, Point, RequestPopupModelAction, SetPopupModelAction, UpdateModelAction } from 'sprotty-protocol';
 
-/**
- * Extends Sprotty's SLabel to add support for having the initial text when entering
- * in direct edit mode different from the text's label itself, and makes the
- * pre-selection of the edited text optional.
- */
-export class SEditableLabel extends SLabel {
-  initialText: string;
-  preSelect: boolean = true;
-}
-
 class EditLabelUIWithInitialContent extends EditLabelUI {
   protected applyTextContents() {
-    if (this.label instanceof SEditableLabel) {
-      this.inputElement.value = this.label.initialText || this.label.text;
+    if (this.label instanceof Label) {
```

Comment:
This class is starting to get big, could you extract it in a dedicated file please.

Reply:
move to EditLabelUIWithInitialContent  class